doc: update documentation for 2.7.99-cs1 (#16337)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-2.7.99-cs1.rst

Changelog
*********

The following sections provide detailed lists of changes by component.

Zephyr
======

.. NOTE TO MAINTAINERS: All the Zephyr commits in the below git commands must be handled specially after each upmerge and each nRF Connect SDK release.

The Zephyr fork in |NCS| (``sdk-zephyr``) contains all commits from the upstream Zephyr repository up to and including ``28b6861211d2d9f54411ff60357a42d84a253600``, with some |NCS| specific additions.

* Added:

  * Experimental support for Power management with support for System On All Idle power state.
  * Experimental support for a new peripheral driver SAADC.
    See the :zephyr:code-sample:`adc_sequence` and :zephyr:code-sample:`adc_dt` sample documentation for more details.
  * Experimental support for the clock control API.
    See the :file:`include/zephyr/drivers/clock_control/nrf_clock_control.h` file for more details.

For the list of upstream Zephyr commits (not including cherry-picked commits) incorporated into nRF Connect SDK since the most recent release, run the following command from the :file:`ncs/zephyr` repository (after running ``west update``):

.. code-block:: none

>    git log --oneline 28b6861211 ^25e90e7bb0

Reviewer: @gmarull : I have updated the latest commit now. Please confirm this.

.. _ncs_release_notes_2799_cs1:

|NCS| v2.7.99-cs1 Release Notes
###############################

.. contents::
   :local:
   :depth: 3

|NCS| v2.7.99-cs1 is a **customer sampling** release, tailored exclusively for participants in the nRF54H20 SoC customer sampling program.
Do not use this release of |NCS| if you are not participating in the program.

The release is not part of the regular release cycle and is specifically designed to support early adopters working with the nRF54H20 SoC.
It is intended to provide early access to the latest developments for nRF54H20 SoC, enabling participants to experiment with new features and provide feedback.

The scope of this release is intentionally narrow, concentrating solely on the nRF54H20 DK.
You can use the latest stable release of |NCS| to work with other boards.

All functionality related to nRF54H20 SoC is considered experimental.

Highlights
**********

Added the following features as supported:

> * Power management - Support for System On All Idle power state.
> * Clock control API.

Reviewer 2: ```suggestion
* Power management with support for System On All Idle power state.
* Clock control API.
```

Added the following feature as experimental:

> * Support to a new peripheral driver SAADC.

Reviewer 2: ```suggestion
* New peripheral driver SAADC.
```

Release tag
***********

The release tag for the |NCS| manifest repository (|ncs_repo|) is **v2.7.99-cs1**.
Check the :file:`west.yml` file for the corresponding tags in the project repositories.

To use this release, check out the tag in the manifest repository and run ``west update``.
See :ref:`cloning_the_repositories` and :ref:`gs_updating_repos_examples` for more information.

For information on the included repositories and revisions, see `Repositories and revisions for v2.7.99-cs1`_.

IDE and tool support
********************

`nRF Connect extension for Visual Studio Code <nRF Connect for Visual Studio Code_>`_ is the recommended IDE for |NCS| v2.7.99-cs1.
See the :ref:`installation` section for more information about supported operating systems and toolchain.

.. note::

   Because of the significant, potentially breaking changes introduced in the |NCS| v2.7.0, the support for v2.7.99-cs1 in the |nRFVSC| is experimental in the following areas:

>      * sysbuild - During the extension stabilization period, manually edit and verify Kconfig option values instead of using the nRF Kconfig GUI.

Reviewer 2: ```suggestion
     * Sysbuild - During the extension stabilization period, Kconfig option values need to be manually edited and verified instead of using the nRF Kconfig GUI.
```

Reviewer: This follows the same template as in the 2.7 release notes. https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/releases_and_maturity/releases/release-notes-2.7.0.html#ide_and_tool_support. Should this still be changed?

>      * Hardware Model v2 - During the extension stabilization period, edit devicetree files manually instead of using the Devicetree Visual Editor.

Reviewer 2: ```suggestion
     * Hardware Model v2 - During the extension stabilization period, devicetree files need to be edited manually instead of using the Devicetree Visual Editor.
```

Reviewer: same comment as above.

       Moreover, custom boards added in the extension will follow the Zephyr's Hardware Model v1.

   The extension users that need v2.7.99-cs1 should `switch to the pre-release version of the extension`_.

Migration notes
***************

See the `Migration notes for nRF Connect SDK v2.7.99-cs1 and the nRF54H20 DK`_ for the changes required or recommended when migrating your environment to the |NCS| v2.7.99-cs1.

Changelog
*********

The following sections provide detailed lists of changes by component.

Zephyr
======

.. NOTE TO MAINTAINERS: All the Zephyr commits in the below git commands must be handled specially after each upmerge and each nRF Connect SDK release.

The Zephyr fork in |NCS| (``sdk-zephyr``) contains all commits from the upstream Zephyr repository up to and including ``ea02b93eea35afef32ebb31f49f8e79932e7deee``, with some |NCS| specific additions.

* Added:

>   * Power management - Support for System On All Idle power state.
>   * Support to a new peripheral driver SAADC as experimental.

Reviewer 2: ```suggestion
  * Power management - Support for System On All Idle power state.
  * Experimental support for a new peripheral driver SAADC.
```

>   * Support for the Clock control API.

Reviewer 3: See https://github.com/nrfconnect/sdk-nrf/pull/17015/files for reference.
```suggestion
  * Support for the clock control API.
```

For the list of upstream Zephyr commits (not including cherry-picked commits) incorporated into nRF Connect SDK since the most recent release, run the following command from the :file:`ncs/zephyr` repository (after running ``west update``):

.. code-block:: none

   git log --oneline a415e2d702 ^23cf38934c

For the list of |NCS| specific commits, including commits cherry-picked from upstream, run:

.. code-block:: none

   git log --oneline manifest-rev ^a415e2d702

The current |NCS| main branch is based on revision ``a415e2d702`` of Zephyr.

.. note::
   For possible breaking changes and changes between the latest Zephyr release and the current Zephyr version, refer to the :ref:`Zephyr release notes <zephyr_release_notes>`.

Documentation
=============

* Added:

  * A page about :ref:`ug_nrf54h20_architecture_clockman`.
  * A page about :ref:`ug_nrf54h20_architecture_pm`.

>   * A :ref:`multicore_idle_test` sample documentation.

Reviewer 2: ```suggestion
  * Documentation for :ref:`multicore_idle_test`.
```

>   * A :ref:`multicore_idle_test` sample documentation.

Reviewer 2: This entry should move up in the samples section and should be modified into 'Added new sample .. ' under Other samples.

Reviewer: In PR https://github.com/nrfconnect/sdk-nrf/pull/17015, changes are made only to the sample doc. Hence, documentation section could be the better place.

.. _ncs_release_notes_2799_cs1:

|NCS| v2.7.99-cs1 Release Notes
###############################

.. contents::
   :local:
   :depth: 3

|NCS| v2.7.99-cs1 is a **customer sampling** release, tailored exclusively for participants in the nRF54H20 SoC customer sampling program.
Do not use this release of |NCS| if you are not participating in the program.

The release is not part of the regular release cycle and is specifically designed to support early adopters working with the nRF54H20 SoC.
It is intended to provide early access to the latest developments for nRF54H20 SoC, enabling participants to experiment with new features and provide feedback.

The scope of this release is intentionally narrow, concentrating solely on the nRF54H20 DK.
You can use the latest stable release of |NCS| to work with other boards.

All functionality related to nRF54H20 SoC is considered experimental.

Highlights
**********

> Added the following features as supported:

Reviewer 4: those features are experimental

* Power management with support for System On All Idle power state.
* Clock control API.

> Added the following feature as experimental:

Reviewer 4: all is experimental

Changelog
*********

The following sections provide detailed lists of changes by component.

Zephyr
======

.. NOTE TO MAINTAINERS: All the Zephyr commits in the below git commands must be handled specially after each upmerge and each nRF Connect SDK release.

The Zephyr fork in |NCS| (``sdk-zephyr``) contains all commits from the upstream Zephyr repository up to and including ``a415e2d702d5192ffdf7ddf67fa73f8fab70a277``, with some |NCS| specific additions.

* Added:

  * Power management - Support for System On All Idle power state.
  * Experimental support for a new peripheral driver SAADC.
    See the :zephyr:code-sample:`adc_sequence` and :zephyr:code-sample:`adc_dt` sample documentation for more details.
  * Support for the clock control API.
    See the :file:`include/zephyr/drivers/clock_control/nrf_clock_control.h` file for more details.

For the list of upstream Zephyr commits (not including cherry-picked commits) incorporated into nRF Connect SDK since the most recent release, run the following command from the :file:`ncs/zephyr` repository (after running ``west update``):

.. code-block:: none

>    git log --oneline a415e2d702 ^e7415e4e39

Reviewer: @gmarull : Please check whether the commit "e7415e4e39" is correct.

Reviewer 5: looks good

.. _ncs_release_notes_2799_cs1:

|NCS| v2.7.99-cs1 Release Notes
###############################

.. contents::
   :local:
   :depth: 3

|NCS| v2.7.99-cs1 is a **customer sampling** release, tailored exclusively for participants in the nRF54H20 SoC customer sampling program.
Do not use this release of |NCS| if you are not participating in the program.

The release is not part of the regular release cycle and is specifically designed to support early adopters working with the nRF54H20 SoC.
It is intended to provide early access to the latest developments for nRF54H20 SoC, enabling participants to experiment with new features and provide feedback.

The scope of this release is intentionally narrow, concentrating solely on the nRF54H20 DK.
You can use the latest stable release of |NCS| to work with other boards.

All functionality related to nRF54H20 SoC is considered experimental.

Highlights
**********

> This release introduces following features:

Reviewer 2: Could we have this modified as
```suggestion
Added the following features as supported:
```
similar to the entries in the older release notes.

* Power management - Support for System On All Idle power state.

> * Support to a new peripheral driver SAADC as experimental.

Reviewer 2: Probably could be under: Added the following features as experimental:

* Clock control API.

> 

Reviewer: @kato-nordic - Please check the release highlights.

.. note::

   Because of the significant, potentially breaking changes introduced in the |NCS| v2.7.0, the support for v2.7.99-cs1 in the |nRFVSC| is experimental in the following areas:

     * sysbuild - During the extension stabilization period, manually edit and verify Kconfig option values instead of using the nRF Kconfig GUI.
     * Hardware Model v2 - During the extension stabilization period, edit devicetree files manually instead of using the Devicetree Visual Editor.
       Moreover, custom boards added in the extension will follow the Zephyr's Hardware Model v1.

   The extension users that need v2.7.99-cs1 should `switch to the pre-release version of the extension`_.

Migration notes
***************

See the :ref:`migration_guides` page for a set of miration guides specifically tailored for the nRF54H20 customer sampling paritipants.

Changelog
*********

The following sections provide detailed lists of changes by component.

TBD

Zephyr
======

.. NOTE TO MAINTAINERS: All the Zephyr commits in the below git commands must be handled specially after each upmerge and each nRF Connect SDK release.

> The Zephyr fork in |NCS| (``sdk-zephyr``) contains all commits from the upstream Zephyr repository up to and including ``ea02b93eea35afef32ebb31f49f8e79932e7deee``, with some |NCS| specific additions.

Reviewer: @rlubos , @shanthanordic - Could you please confirm whether the zephyr commit and below details are correct.

Reviewer: @gmarull : Since Robert lubos is OOO, could you please confirm whether the zephyr commit and below details are correct.

Reviewer 5: If this targets main, it is currently `a415e2d702d5192ffdf7ddf67fa73f8fab70a277`. Just check `west.yml` at the point of tagging.
